compiler/src/dotty/tools/dotc/reporting/MessageKind.scala

```diff
@@ -0,0 +1,35 @@
+package dotty.tools.dotc.reporting
+
+/** The various message kinds that can be used inside of a Message */
```

Comment:
Maybe expand the documentation to encourage people to add new categories if needed? Otherwise they may feel like they have to choose something in here even if it doesn't fit.

Reply:
Sounds good. I added some further explanation.